### ISSUE TITLE
Compression changes orientation

### ISSUE THREAD

**Issue by nachtmaar** [OpenFlutter/flutter_image_compress#15]
TL;DR
The plugin strips EXIF information when compressing the image (only tested on Android). This changes orientation for my image.
Example project code is [here](https://github.com/nachtmaar/flutter_image_compress_bug).

I have the following bug on Android:
If I use any compression function of FlutterImageCompress with `rotation: 0`, the image is rotated after the compression (although it shouldn't be rotated). In the example where is an image taken with a Samsung GT-I9195.

According to the EXIF information the orientation is 6. This means rotated by 90° counterclockwise.

The original image:
```bash
identify -verbose assets/passierte_tomaten_samsung.jpg|grep -i orientation
  Orientation: RightTop
    exif:Orientation: 6
```

<img width="457" alt="original" src="https://user-images.githubusercontent.com/8013145/48673586-23289b00-eb43-11e8-9b48-b77e353d1e82.png">

The compressed image is missing the orientation in the EXIF information (Android)

```bash
identify -verbose apps/com.example.flutterimagecompressbug/r/app_flutter/flutter_image_compress.jpg |grep -i orientation
  Orientation: Undefined
```

<img width="457" alt="after_compression" src="https://user-images.githubusercontent.com/8013145/48673585-22900480-eb43-11e8-9576-af6dc5982c93.png">

I did not check yet, if the EXIF information are also stripped on iOS.

**Comment by CaiJingLong** [comment]
you can see #3.
You can combine some existing Exif libraries to achieve your goals.
The purpose of this library is just to compress the image. You can get the angle in advance, and then pass the parameters into the function, so that you can get the goal you want in the result.

**Comment by nachtmaar** [comment]
And what about keeping the EXIF information?

**Comment by CaiJingLong** [comment]
Keeping Exif in my view is not the main goal. of this library.
I will take this as a goal, but I will not guarantee the realization time.
If you have the ability, welcome PR.

**Comment by nachtmaar** [comment]
I think it is also a good idea to mention that EXIF information are stripped in the README.

**Comment by CaiJingLong** [comment]
> I think it is also a good idea to mention that EXIF information are stripped in the README.

I will add the EXIF will be removed in README.

**Comment by crushman1** [comment]
As there is no fix to this, i would like a way to simply provide the end user the ability to rotate the image themselves. i have been searching all over the web but cannot figure out how to accomplish this. Can someone please point me in the right direction? Lives Are At Stake >.<

**Comment by CaiJingLong** [comment]
> As there is no fix to this, i would like a way to simply provide the end user the ability to rotate the image themselves. i have been searching all over the web but cannot figure out how to accomplish this. Can someone please point me in the right direction? Lives Are At Stake >.<

If you know your image's orientation , you can use `rotate` params to rotate your image in compress method.

**Comment by Cretezy** [comment]
One possible way to fix this would be to readd the EXIF data after the resizing, using the `image` package.

**Comment by crushman1** [comment]
> One possible way to fix this would be to readd the EXIF data after the resizing, using the `image` package.

You mean "before" the resizing right? from what i understand, the plugin strips out the EXIF data, doesnt it?

**Comment by Cretezy** [comment]
@crushman1 

I mean like:
```
- save rotation exif data
- do compression
- restore rotation exif data
```

**Comment by CaiJingLong** [comment]
@Cretezy 

On the [exif](https://github.com/OpenFlutter/flutter_image_compress/tree/exif) branch, I used a third-party library a week ago to implement what you said.
Using this method, I can read EXIF by exif reader, but the orientation attribute does not actually take effect when browsing.

Because I don't know enough about EXIF info, this issue can only be shelved temporarily.

**Comment by juliengit2** [comment]
pity that exif data are not kept

**Comment by msw333** [comment]
That's really a pity and makes this package pretty unusable for me

**Comment by richie-south** [comment]
Anybody got a working "workaround" for this problem? :) I understand that this library doesn't want to auto rotate images.

But did anybody get this idea to work? 😃 
```
- save rotation exif data
- do compression
- restore rotation exif data
```

**Comment by crushman1** [comment]
> Anybody got a working "workaround" for this problem? :) I understand that this library doesn't want to auto rotate images.
> 
> But did anybody get this idea to work? 😃
> 
> ```
> - save rotation exif data
> - do compression
> - restore rotation exif data
> ```

@richie-south The work-around i got was to not use the plugin at all.  To compress images, I set the maxheight and max width using the[ Image picker](https://pub.dev/packages/image_picker) plugin like so : 
`ImagePicker.pickImage(source: ImageSource.camera, maxHeight: 600, maxWidth: 600)`
The smaller the image, the smaller the file.

**Comment by richie-south** [comment]
@crushman1 Okej thanks for the response and example 😃 i might have go with that approach to :/, really want to compress the file sizes more.

**Comment by CaiJingLong** [comment]
The library is solved the problem of angle. You just use [autocorrectionangle](https://github.com/OpenFlutter/flutter_image_compress#autocorrectionangle) and the param default is used.

The problem of direction was solved, and the issue why it was not closed was that Exif's information was not completely preserved.

**Comment by CaiJingLong** [comment]
I update some code to keep EXIF. 
You can use git ref try to import the change.

```yaml
flutter_image_compress:
  git: 
    url: https://github.com/OpenFlutter/flutter_image_compress.git
    ref: 733e369bbb9154541cc1474ab9970f38bd506353
```

**Comment by CaiJingLong** [comment]
Update it in 0.6.1.
Use param `keepExif` to keep EXIF.
